### PR TITLE
feat: add cors to script and preload tags

### LINKED ISSUE #6634: nuxt.config.js build.crossorigin with nuxt generate not working
### Version

[v2.10.2](https://github.com/nuxt.js/releases/tag/v2.10.2)

### Reproduction link

[https://github.com/nuxt/nuxt.js/blob/dev/packages/vue-renderer/src/renderers/ssr.js#L157](https://github.com/nuxt/nuxt.js/blob/dev/packages/vue-renderer/src/renderers/ssr.js#L157)

### Steps to reproduce

see the code,
nuxt use vue-server-renderer's renderScripts methods, but there is noway to set custom attributes in it, because the script tag is hard coded to only has src and defer attributes
see: https://github.com/vuejs/vue/blob/dev/packages/vue-server-renderer/build.dev.js#L9041
and I post a issue to vue: https://github.com/vuejs/vue/issues/10780 to support custom attributes
so the build.crossorigin property won't work in generated html files

### What is expected ?

build.crossorigin work in generated html files, and the script will has crossorigin attributes

### What is actually happening?

the generated files are like these
```
<script src="https://xxx/js/e5e9af4806e44b81521f.js" defer></script><script src="https://xxx/js/6d0f75fe77a36f6e0dfa.js" defer></script><script src="https://xxx/js/d7b711a83c12b9d56535.js" defer></script>
...
```

<div align="right"><sub><em>This bug report is available on <a href="https://cmty.app/nuxt">Nuxt</a> community (<a href="https://cmty.app/nuxt/nuxt.js/issues/c9960">#c9960</a>)</em></sub></div>

### PR DESCRIPTION
**Pull request by hxlniada**
## Types of changes

- [ x] Bug fix (a non-breaking change which fixes an issue)
- [ ] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Description
Resolves: #6634 

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly. (PR: #)
- [ ] I have added tests to cover my changes (if not applicable, please state why)
- [x] All new and existing tests are passing.

### CODE REVIEW

File: packages/vue-renderer/src/renderers/ssr.js [hunk 1]
```diff
@@ -20,15 +20,36 @@ export default class SSRRenderer extends BaseRenderer {
   }
 
   renderScripts (renderContext) {
-    return renderContext.renderScripts()
+    let scripts = renderContext.renderScripts()
+    const scriptPattern = /<script[^>]*?src="([^"]*?)"[^>]*?>[^<]*?<\/script>/g
+
+    scripts = scripts.replace(scriptPattern, (scriptTag, jsFile) => {
```
Comment (pi0) [inline packages/vue-renderer/src/renderers/ssr.js]: Can we refactor this functionality (parts repeated in `renderScripts` and `renderResourceHints`)

**Review by pi0 — CHANGES_REQUESTED** [review]

**Review by clarkdo — COMMENTED** [review]
I think replacing script for all SRR rendering is kind of expensive for app runtime.

We'd better open a pr in Vue to change `renderScripts` and `renderResourceHints`, make the support attributes, or at least delegate them in nuxt.

If you don't have enough time on it, I can open a pr and let's close this one.

**Comment by hxlniada** [comment]
@clarkdo OK, I agree with you, and i do not have enough time on it

**Comment by clarkdo** [comment]
Cool, I’ll work on this

**Comment by Austio** [comment]
@clarkdo, this PR seems good to go but wanted to check if there was anything i can do to help get over the line and merged?  Thanks!

**Comment by clarkdo** [comment]
@Austio Sorry, I’ll try to find some times to work on it this week.

**Comment by Austio** [comment]
@clarkdo no need to be sorry, any way that i can help?

**Comment by clarkdo** [comment]
@Austio I'll try to make a pr this week, if you can test with your app, it would be great!

**Comment by clarkdo** [comment]
As Vue 2 won't be release frequently, so let's fix this issue in Nuxt for now, I'll open a new pr.

**Pull request closed without merging**